include/audio/aseq.h

```diff
@@ -0,0 +1,2341 @@
+    _wr_lbl     \label
+.endm
+
+/**
+ * samplebook <value:u8>
+ *
+ *  Sets the sample book mode.
+ */
+.macro samplebook value
+    _wr_cmd_id  samplebook, ,0xE6,,,,,,, 0, 0
+    _wr_u8      \value
+.endm
+
+/**
+ * reverbidx <arg:u8>
+ *
+ *  Sets the channel reverb.
+ */
+.macro reverbidx arg
+    _wr_cmd_id  reverbidx, ,0xE5,,,,,,, 0, 0
+    _wr_u8      \arg
+.endm
+
+/**
+ * reverbidx <arg:u8>
+ *
+ *  Sets the channel vibrato delay.
+ */
+.macro vibdelay arg
+    _wr_cmd_id  vibdelay, ,0xE3,,,,,,, 0, 0
+    _wr_u8      \arg
+.endm
+
+/**
+ * vibdepthgrad <extentStart:u8> <extentTarget:u8> <extentChangeDelay:u8>
+ *
+ *  Sets the vibrato extent.
+ */
+.macro vibdepthgrad arg0, arg1, arg2
+    _wr_cmd_id  vibdepthgrad, ,0xE2,,,,,,, 0, 0
+    _wr_u8      \arg0
+    _wr_u8      \arg1
+    _wr_u8      \arg2
+.endm
+
+/**
+ * vibfreqgrad <rateStart:u8> <rateTarget:u8> <rateChangeDelay:u8>
+ *
+ *  Sets the vibrato rate.
+ */
+.macro vibfreqgrad arg0, arg1, arg2
+    _wr_cmd_id  vibfreqgrad, ,0xE1,,,,,,, 0, 0
+    _wr_u8      \arg0
+    _wr_u8      \arg1
+    _wr_u8      \arg2
+.endm
+
+/**
+ * volexp <amt:u8>
+ *
+ *  Changes the expression amount for the channel.
+ */
+.macro volexp amt
+    _wr_cmd_id  volexp, ,0xE0,,,,,,, 0, 0
+    _wr_u8      \amt
+.endm
+
+/**
+ * transpose <semitones:s8>
+ *
+ *  Sets the transposition amount. Transposition shifts all note pitches by the
+ *  provided number of semitones.
+ */
+.macro transpose semitones
+    _wr_cmd_id  transpose, 0xDF,0xDB,0xC2,,,,,, 0, 0
+    _wr_s8      \semitones
+.endm
+
+/**
+ * rtranspose <semitones:s8>
+ *
+ *  Adjusts the transposition amount. This is only available at the top sequence level.
+ */
+.macro rtranspose semitones
+    _wr_cmd_id  rtranspose, 0xDE,,,,,,,, 0, 0
+    _wr_s8      \semitones
+.endm
+
+/**
+ * freqscale <arg:s16>
+ *
+ *  Sets the freqScale for the current channel.
+ */
+.macro freqscale arg
+    _wr_cmd_id  freqscale, ,0xDE,,,,,,, 0, 0
+    _wr_s16     \arg
+.endm
+
+/**
+ * tempo <bpm:u8>
+ *
+ *  Changes the tempo of the sequence.
+ */
+.macro tempo bpm
+    _wr_cmd_id  tempo, 0xDD,,,,,,,, 0, 0
+    _wr_u8      \bpm
+.endm
+
+/**
+ * tempochg <arg:s8>
+ *
+ *  Sets the tempoChange for the sequence.
+ */
+.macro tempochg arg
+    _wr_cmd_id  tempochg, 0xDC,,,,,,,, 0, 0
+    _wr_s8      \arg
+.endm
+
+/**
+ * pan <amt:u8>
+ *
+ *  Changes the pan amount for a channel.
+ */
+.macro pan pan
+    /* pan can only take values in 0..127 */
+    _check_arg_bitwidth_u \pan, 7
+    _wr_cmd_id  pan, ,0xDD,,,,,,, 0, 0
+    _wr_u8      \pan
+.endm
+
+/**
+ * panweight <weight:u8>
+ *
+ *  Controls how much the final pan value is influenced by the channel pan and layer pan.
+ *  The layer pan is set by the drum instrument pan.
+ *  A value of 0 ignores channel pan, while a value of 128 ignores layer pan.
+ *
+ *      finalPan = (weight * channelPan + (128 - weight) * layerPan) / 128
+ */
+.macro panweight weight
+    /* pan can only take values in 0..127 */
```

Comment:
`pan` -> `weight`? Also this seems to contradict the description, where a value of 128 is mentioned